### PR TITLE
Upgrade embedded GLEW to version 1.13.0

### LINKED ISSUE #5087: Update embedded GLEW files
We have a couple GLEW files in our `drivers/gl_context`:
- [glew.c](https://github.com/godotengine/godot/blob/master/drivers/gl_context/glew.c)
- [glew.h](https://github.com/godotengine/godot/blob/master/drivers/gl_context/glew.h)
- [glxew.h](https://github.com/godotengine/godot/blob/master/drivers/gl_context/glxew.h)
- [wglew.h](https://github.com/godotengine/godot/blob/master/drivers/gl_context/wglew.h)

I don't know yet how old those files are, but at least pre-open sourcing (Feb 2014), so they could use an update. We did some patches here and there to handle build issues on some platforms, so hopefully upstream development would have made them non necessary after we upgrade.

### PR DESCRIPTION
**Pull request by akien-mga**
Also drop all our customisations of the upstream files, by:
- Fixing the buggy Haiku setup (GLEW usage on Haiku should be investigated by users of this platform, grepping for `__HAIKU__` does give matches in the GLEW code, so it might be supported now)
- Use SCons magic to make a static build, instead of hardcoding it in the source files
- Also relies on SCons not to build `glew.c` when `GLEW_ENABLED` is not defined

Fixes #5087.

### CODE REVIEW

**Comment by vnen** [comment]
Fails to compile on MSVC: `.\drivers/gles2/rasterizer_gles2.h(51): fatal error C1083: Cannot open include file: 'gl_context/glew.h': No such file or directory`

**Comment by akien-mga** [comment]
Thanks, I've fixed that locally, I'll soon push a rebased version.

I also noticed that I had forgotten to readd the `#ifdef GLEW_ENABLED` in `glew.c`, so I'm working on a SCons-based variant to have the same behaviour without having to touch the upstream source file.

**Comment by akien-mga** [comment]
Rebased and added a commit for the `GLEW_ENABLED` stuff :)

File: drivers/gl_context/SCsub [hunk 1]
```diff
@@ -2,7 +2,8 @@ Import('env')
 Import('env_drivers')
 
 env_drivers.add_source_files(env.drivers_sources,"*.cpp")
-env_drivers.add_source_files(env.drivers_sources,"glew.c")
-
-env_drivers.Append(CFLAGS = ['-DGLEW_STATIC'])
-env_drivers.Append(CPPPATH = ['.'])
+if (env['glew'] == 'yes'):
```
Comment (akien-mga) [inline drivers/gl_context/SCsub]: Apparently that won't work if `env['glew']` was not defined, so I'll have to use `if ('glew' in env and env['glew'] == 'yes'`.

Comment (J08nY) [inline drivers/gl_context/SCsub]: Isn't SConscript essentially python? In that case, if `env` is a python `dict`you can use `env.get('glew')` which returns `None` if it's not there and the value if it is.

Comment (akien-mga) [inline drivers/gl_context/SCsub]: Thanks, I'll use that.

**Pull request merged**